docs/content/docs/data-structures/map.md

```diff
@@ -1,10 +1,449 @@
+
+### Using the DDS API
+
+A `SharedMap` instance can also be created from within your own custom data object. Each extension of the `DataObject` class has access to its own `runtime` that manages the object's own data store. `SharedMap` provides a static `create` function on its API that accepts this runtime and will provide you with a new instance of the DDS.
+
+```javascript
+const newMap = SharedMap.create(this.runtime, id);
+```
+
+`DataObject` classes also provide by default a `root` [SharedDirectory]({{< relref "directory.md" >}}) object that is always available and provides a location to store the map. The `SharedMap` itself also provides a [handle]({{< relref "data-modeling.md#using-handles-to-store-and-retrieve-fluid-objects" >}}) property that can be thought of as a serializable form of the map. Now that you have a way to serialize the map and a place where to put it, you can do the following to store the map in the `root`:
+```javascript
+const newMapKey = "uniqueMapId";
+this.root.set(newMapKey, newMap.handle);
+```
+And then, you can load it back in the following manner:
+```javascript
+const newMapHandle = this.root.get(newMapKey);
+const newMapLoaded = await newMapHandle.get();
+```
+`newMapLoaded` now points to the same DDS as your initial `newMap`
+
+## API Functionality
+
+The `SharedMap` object provides a number of functions to allow you to edit the key/value pairs stored on the object. As stated earlier, these are intended to match the `Map` API. However, the keys used in `SharedMap` must be strings. Each edit will also trigger a `valueChanged` event which will be discussed in the **Events** section below.
+
+- `set(key, value)` - Used for updating the value stored at `key` with the new provided value
+- `get(key)` - Returns the latest value stored on the key or `undefined` if it does not exist
+- `has(key)` - Checks to see if the key is available. This can be used as a boolean check prior to using `get` to avoid returning an `undefined` value
+- `keys()` - Returns all the keys that have been set for this map
+- `entries()` - Returns an iterator for all values stored on the map. This can be easily converted into an array by doing `Array.from(map.entries())`
+- `delete(key)` - Removes the key from the map
+{{< callout note >}}
+
+When removing a key from the `SharedMap`, please use the `delete` function instead of using `set` with `value` as `undefined`. The latter will not remove the key from the object and it will still be returned when developers call `keys` or `entries`
+
+{{< /callout >}}
```

Comment:
Committed in the earlier suggestion to include tips in the introduction to distinguish between Map and SharedMap. Also, I think developers may have their own reasons for storing an undefined value in the SharedMap (i.e. perhaps the  key is set but the value is still being loaded due to an async API call and its undefined until the async call finishes), similar to how they could store it into a Map. I was just trying to specify that they should call `delete` if they want to remove the value entirely, but perhaps that's unnecessary since that behavior would be the same as using Map itself